### PR TITLE
Advanced Search : Handle invalid range in date input

### PR DESCRIPTION
**Pull request by olamothe**
This will display an error message next to the date picker when the date range is invalid.

Also add a breadcrumb item to easily clear filters in advanced search.

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: sass/_SearchInterface.scss [hunk 1]
```diff
@@ -36,6 +36,10 @@
   * {
     @include box-sizing(border-box);
   }
+
+  .coveo-error {
```
Comment (francoislg) [inline sass/_SearchInterface.scss]: Could conflict with current implementations, but I would agree that there is a very, _very_ low probability of breaking anything :)

File: src/ui/AdvancedSearch/DateInput/DateInput.ts [hunk 2]
```diff
@@ -35,16 +36,36 @@ export class DateInput implements IAdvancedSearchInput {
...
     }
   }
 
   protected getRadio(): HTMLInputElement {
     return <HTMLInputElement>$$(this.element).find('input');
   }
 
+
+  private showError(message: string) {
+    this.hideError();
```
Comment (francoislg) [inline src/ui/AdvancedSearch/DateInput/DateInput.ts]: Weird to see `this.hideError()` as the first line of `showError`. Should probably be `removeError()` :)

File: src/ui/AdvancedSearch/DateInput/DateInput.ts [hunk 3]
```diff
@@ -35,16 +36,36 @@ export class DateInput implements IAdvancedSearchInput {
...
+      this.showError(error);
     }
   }
 
   protected getRadio(): HTMLInputElement {
     return <HTMLInputElement>$$(this.element).find('input');
   }
 
+
+  private showError(message: string) {
```
Comment (francoislg) [inline src/ui/AdvancedSearch/DateInput/DateInput.ts]: What about `setErrorMessage(message: string)` instead?

File: src/ui/FacetSlider/FacetSlider.ts [hunk 4]
```diff
@@ -618,7 +618,7 @@ export class FacetSlider extends Component {
 
   private buildSlider() {
     let sliderContainer = $$('div', {
-      className: 'coveo-facet-values coveo-slider-container'
+      className: 'coveo-slider-container'
```
Comment (francoislg) [inline src/ui/FacetSlider/FacetSlider.ts]: Just double check that `coveo-facet-values` removal. Low risk IMO

File: test/ui/DatePickerTest.ts [hunk 5]
```diff
@@ -21,6 +21,10 @@ export function DatePickerTest() {
       expect(element.readOnly).toBe(true);
     });
```
Comment (francoislg) [inline test/ui/DatePickerTest.ts]: Some missing tests :slightly_smiling_face: 

- `getDateValue`
- `this.onChange(this);` is now in `reset()`

File: src/ui/AdvancedSearch/DateInput/DateInput.ts [hunk 6]
```diff
@@ -35,16 +36,36 @@ export class DateInput implements IAdvancedSearchInput {
   }
 
   public updateQuery(queryBuilder: QueryBuilder) {
-    let value = this.getValue();
-    if (value) {
-      queryBuilder.advancedExpression.add(this.getValue());
+    try {
+      let value = this.getValue();
+      if (value) {
+        queryBuilder.advancedExpression.add(this.getValue());
```
Comment (dms1lva) [inline src/ui/AdvancedSearch/DateInput/DateInput.ts]: Do we need to call `getValue` again here when update the advanced expression? Also, the `DateInput` class on its own makes no sense. The `getValue` method returns an empty string which would make the `updateQuery` method dead code. I think it should be an abstract class and the `getValue` method should be abstract.

File: src/ui/AdvancedSearch/DateInput/BetweenDateInput.ts [hunk 7]
```diff
@@ -31,20 +33,32 @@ export class BetweenDateInput extends DateInput {
   }
 
   public getValue(): string {
-    let firstDate = this.firstDatePicker.getValue();
-    let secondDate = this.secondDatePicker.getValue();
+    const firstDate = this.firstDatePicker.getDateValue();
```
Comment (olamothe) [inline src/ui/AdvancedSearch/DateInput/BetweenDateInput.ts]: firstDateAsString is the date value as a string for the query,

firstDate is the date value as a actual date object to do timespan comparison (and check if the timespan is invalid)

**Pull request merged**